GEMB/managelayers.m

```diff
@@ -35,19 +35,23 @@
 n=length(T);
 zY2=zY;
 dzMin2=zeros(size(dz));
+dzMax2=zeros(size(dz));
 
 X=1;
 Zcum = cumsum(dz); 
+
 % check if depth is too small
-dzMin2(1)=dzMin;
-for i=2:n
-    if (Zcum(i)<=zTop+Dtol)
-        dzMin2(i)=dzMin;
-        X=i;

```

Comment:
I think this change might introduce errors. Previously, after `X=1` was defined, it could get updated in the loop by `X=i`, but that loop has now been removed. This means that further down in the code when we evaluate
```matlab 
if (i<=X && dz(i)<dzMin-Dtol) || (i>X && dz(i)<dzMin2(i)-Dtol)
```
the variable `X` will always have a value of 1. 

I would prefer to eliminate the `X` variable entirely, and instead use a logical index like: 
```matlab 
is_top_layer = Zcum <= (zTop + Dtol); 
```